leasing/serializers/land_use_agreement.py

```diff
@@ -523,7 +534,376 @@ def update(self, instance, validated_data):
+        invoice_row = super().create(validated_data)
+
+        invoice_row.update_amount()
+        invoice_row.save()
+
+        return invoice_row
+
+    def validate(self, data):
+        """Validate that rows with an inactive receivable type cannot be created
+
+        Saving an existing row should succeed, but creating
+        new rows or changing a rows receivable type to an inactive type
+        should fail."""
+
+        valid = True
+
+        if not data["receivable_type"].is_active:
+            if self.instance:
+                if self.instance.receivable_type != data["receivable_type"]:
+                    # We have an existing row but the receivable type wasn't the
+                    # same as before
+                    valid = False
+            else:
+                # We have data without an instance.
+                # If there is no id it's a new row.
+                if "id" not in data:
+                    valid = False
+                else:
+                    # Else try to see if the existing row has the same receivable type
+                    try:
+                        existing_row = LandUseAgreementInvoiceRow.objects.get(
+                            pk=data["id"]
+                        )
+
+                        if existing_row.receivable_type != data["receivable_type"]:
+                            valid = False
+                    except ObjectDoesNotExist:
+                        valid = False
+
+        if not valid:
+            raise ValidationError(_("Cannot use an inactive receivable type"))
+
+        return data
+
+
+class LandUseAgreementInvoicePaymentCreateUpdateSerializer(
+    FieldPermissionsSerializerMixin, serializers.ModelSerializer
+):
+    id = serializers.IntegerField(required=False)
+
+    class Meta:
+        model = LandUseAgreementInvoicePayment
+        exclude = ("invoice", "deleted")
+
+
+class LandUseAgreementInvoiceCreateSerializer(
+    UpdateNestedMixin,
+    EnumSupportSerializerMixin,
+    FieldPermissionsSerializerMixin,
+    serializers.ModelSerializer,
+):
+    id = serializers.ReadOnlyField()
+    recipient = InstanceDictPrimaryKeyRelatedField(
+        instance_class=Contact,
+        queryset=Contact.objects.all(),
+        related_serializer=ContactSerializer,
+        required=False,
+    )
+    rows = LandUseAgreementInvoiceRowCreateUpdateSerializer(many=True)
+    payments = LandUseAgreementInvoicePaymentCreateUpdateSerializer(
+        many=True, required=False, allow_null=True
+    )
+    # Make total_amount, billed_amount, and type not required in the serializer and set them in create() if needed
+    total_amount = serializers.DecimalField(
+        max_digits=10, decimal_places=2, required=False
+    )
+    billed_amount = serializers.DecimalField(
+        max_digits=10, decimal_places=2, required=False
+    )
+    type = EnumField(enum=InvoiceType, required=False)
+
+    def override_permission_check_field_name(self, field_name):
```

Comment:
Can this be removed or is it "work in progress"?

Reply:
I keep it still there as I have to still compare to MVJ's invoice code.